perfmetrics/scripts/load_tests/python/load_generator/load_generator.py

```diff
@@ -0,0 +1,357 @@
+# Copyright 2023 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Load Generator for performing load test with a given task (in python).
+
+Contains the class for generating load with the help of a task written in
+python.
+
+Example:
+
+    task_obj = ReadTask()
+    lg_obj = LoadGenerator(...)
+    lg_obj.pre_load_task(...)
+    observations = lg_obj.generate_load(task_obj)
+    metrics = lg_obj.post_load_task(observations, ...)
+"""
+import logging
+import sys
+import threading
+import time
+import json
+import multiprocessing
+import numpy as np
+from load_generator import constants as lg_const
+
+
+class LoadGenerator:
```

Comment:
We can also rename it as PythonIOLoadTester/Performer or similar that will inline with its exposed method.

Reply:
I haven't added Python because this complete code is in 'python' directory and haven't added IO because the load generator can be used with any task and not just the IO tasks. 
Let me know if you think otherwise or if I am missing something.